db/migrate/20170605090919_add_expression_type_to_miq_alert.rb

```diff
@@ -0,0 +1,23 @@
+class ::MiqExpression; end
+
+class AddExpressionTypeToMiqAlert < ActiveRecord::Migration[5.0]
+  class MiqAlert < ActiveRecord::Base
+    serialize :expression
+  end
+
+  def up
+    add_column :miq_alerts, :expression_type, :string
+
+    say_with_time('Add expression type to MiqAlert for existing alerts') do
+      MiqAlert.all.each do |alert|
+        alert.expression_type = Hash if alert.expression.kind_of?(Hash)
+        alert.expression_type = MiqExpression if alert.expression.kind_of?(MiqExpression)
```

Comment:
If we have Ruby objects stored inside the serialized YAML, those are *hard* to solve in migrations.  Easiest solution tends to be to treat the column like a regular string in the stub, and then just look for keywords in the string.  For example:

```ruby
class AddExpressionTypeToMiqAlert < ActiveRecord::Migration[5.0]
  class MiqAlert < ActiveRecord::Base
  end

  def up
    ...
    alert.expression_type = alert.expression.include?("MiqExpression") : "MiqExpression" : "Hash"
    ...
  end
```

If you end up doing the string inclusion style, then you can do this with update_all in two calls:

```ruby
def up
  ...
  MiqAlert.where("expression LIKE ?", "%MiqExpression%").update_all(:expression_type => "MiqExpression")
  MiqAlert.where(:expression_type => nil).update_all(:expression_type => "Hash")
end
```